core/azure-core/src/main/java/com/azure/core/implementation/configuration/Configuration.java

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.implementation.configuration;
+
+import com.azure.core.implementation.logging.ServiceLogger;
+import com.azure.core.implementation.util.ImplUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * Contains configuration information that is used during construction of client libraries.
+ */
+public class Configuration implements Cloneable {
```

Comment:
You should aim for the relevant aspects of this API to be out of the implementation package.